### ISSUE TITLE
react/jsx-no-literals: Add config to allow a set of strings

### ISSUE THREAD

**Issue by benhollander** [yannickcr/eslint-plugin-react#2377]
I propose adding a config option to allow a specified list of strings to pass the `react/jsx-no-literals` rule. I'm finding myself writing code similar to the below pretty often.

## Semi-contrived examples

```javascript

// This would throw an error because of "&nbsp;-&nbsp;"
// Can't do {'&nbsp;-&nbsp;'} because it will be rendered as text instead of an HTML entity
export const DisplayRangeComponent = (props) => (
  <div>
    {/* eslint-disable react/jsx-no-literals */}
    {props.min}&nbsp;-&nbsp;{props.max}
    {/* eslint-enable react/jsx-no-literals */}
  </div>
);

// This would throw because of '*'
export const AddAsterisk = ({ children }) => (
  {/* eslint-disable react/jsx-no-literals */}
  <span>{children}*</span>
  {/* eslint-enable react/jsx-no-literals */}
);
```

It could be convenient (at least in my use-case, which doesn't seem like it would be all that uncommon) to pass in a config like 
```
rules: { 'react/jsx-no-literals': [ 'error', { ignoreStrings: [ '&nbsp;', '*' ] } ] }
```

**Comment by ljharb** [comment]
Is there a reason you can't do: `{this.props.min}{'&nbsp;-&nbsp;'}{this.props.max}`?

**Comment by benhollander** [comment]
> Is there a reason you can't do: `{this.props.min}{'&nbsp;-&nbsp;'}{this.props.max}`?

That will cause the `&nbsp;` to be displayed in the browser rather than be escaped.

I made a codepen to maybe illustrate the problem more clearly: https://codepen.io/bhollander/pen/zgmGyj

![image](https://user-images.githubusercontent.com/1233977/62807241-82a44d80-baba-11e9-92ff-24cbf3496e87.png)

**Comment by ljharb** [comment]
ah right, then `&nbsp;{'-'}&nbsp;`?

**Comment by benhollander** [comment]
That would still fail the rule :sweat_smile:

**Comment by ljharb** [comment]
Gotcha.

In that case, allowing an array of strings to ignore seems reasonable.
